projecttacoma/synthea
# Added Population Attributes to EXM Module (to be used with Keep Modules)

This PR updates the JSON file for the EXM130 module (`EXM130-8.0.000-r4.json`) with boolean attributes that flag the patient as being in the numerator, denominator, and/or initial patient population (IPP), depending on what encounter/procedure states the patient reaches in the module. 

These new attributes allow for the use of keep modules for the numerator/denominator/IPP, which can be specified when patients are created. This will allow the user to only keep patients that belong to the desired patient population.

To test,
- Go into `synthea.properties` in `synthea/src/main/resources` and edit line 64, `generate.log_patients.detail = simple` to `generate.log_patients.detail = detailed`. This will allow for detailed information for each patient to appear in the terminal.
- Run `./run_synthea -p 3 -m "EXM130*-R4"` (or any other number of patients using the `-p` flag) and locate the output for `EXM130-8.0.000-R4 Module`. It contains a list of all the attributes that were set within the module. Look for either `Set_Numerator` or `Set_Denominator` and `Set_IPP` (since both numerator/denominator patients are in the IPP for EXM130). This means the attribute was assigned for the patient. Check that this makes sense based on the module requirements. For example, if the patient had a Procedure and an IPP Qualifying Encounter, the patient should be in the numerator. If the patient had an IPP Qualifying Encounter and no Procedure, the patient should be in the denominator. Otherwise, the patient should have simply reached the terminal state without having any population attributes set.
- Run `./run_synthea -p 3 -m "EXM130*-R4" -k "src/main/resources/keep_modules/keep_numer.json"` (or any other number of patients using the `-p` flag) and check that all the patients' details contain a `Set_Numerator` state. Repeat the same for `keep_denom.json` and `keep_ipp.json` (again, both numerator/denominator patients are in the IPP, so the `keep_ipp.json` results may have a combination of numerator and denominator patients). For reference, the `-k` flag signifies the use of a keep module, and the relevant keep modules are located in `src/main/resources/keep_modules`.
